File: [Manya]/Leetcode/Sorting/948.java
package Sorting;

import java.util.Arrays;

class Solution {
    public int bagOfTokensScore(int[] tokens, int power) {
        Arrays.sort(tokens);
        int start = 0;
        int end = tokens.length-1;
        int score = 0;
        int maxScore = 0;
        while(start <= end){
            if(tokens[start] <= power){
                score++;
                power -= tokens[start];
                start++;
                maxScore = Math.max(score, maxScore);
            }
            
            else if(score > 0){
                power += tokens[end];
                score--;
                end--;
            }
            else{
                break;
            }
        }
        return maxScore;
    }
}

File: Aditya-Pratap/Sorting/948.py
class Solution:
    def bagOfTokensScore(self, tokens: List[int], power: int) -> int:
        score=0
        tokens.sort()
        n=len(tokens)
        i,j=0,n-1
        while i<=j:
            if power>=tokens[i]:
                power-=tokens[i]
                score+=1
                i+=1
            elif score>0 and j-i>1:
                power+=tokens[j]
                score-=1
                j-=1
            else:
                break
        return score
            